### PR TITLE
Feature/norbertspiess

### PR DESCRIPTION
**Pull request by norbertspiess**
Solution for the assignment by Norbert Spiess:

I got confused in the beginning of what a district is. Correlating it more towards the postal communities, although they are extracted from the political data (via the "UML" diagram somehow with visual left to right associations…).
That derailed me quite a lot with my first solution. It was based on raw correlation mappings (thinking in nosql somehow), resolving them within the calculations through referencing values.
Solving it this way took quite some time and was rather ugly. So I decided to have a look again.

After identifying the "real" relations of all entities to each other, I updated the model with composition in mind (basically starting from scratch with commit 683d504100c856013a642a2c9a4499a4d6ec4792).

I decided to keep the original commits and not hide my initial attempt.

In the readme you can find some additional calculations/findings (assessment 5), I showcased.

### CODE REVIEW

File: src/main/java/ch/aaap/assignment/model/PostalCommunity.java [hunk 1]
```diff
@@ -2,11 +2,10 @@
...
 public interface PostalCommunity {
 
-  public String getZipCode();
+  String getZipCode();
 
-  public String getZipCodeAddition();
+  String getZipCodeAddition();
 
-  public String getName();
+  String getName();
```
Comment (robertblust) [inline src/main/java/ch/aaap/assignment/model/PostalCommunity.java:9]: I like how you implemented all relations. What is the unique identifier of a PostalCommunity?

`zipCode + zipCodeAddition`

When you take care about this. You could as well define the PoliticalCommunity on PostalCommunity

`PoliticalCommunity getPoliticalCommunity(); `

**Review by robertblust — COMMENTED** [review]
@norbertspiess Thanks for submitting the Solution. Well done 👍 

According your comment 
> I got confused in the beginning of what a district is. Correlating it more towards the postal communities, although they are extracted from the political data (via the "UML" diagram somehow with visual left to right associations…).

which UML Diagram you refer here?

You will be contacted next week by Simona.

Cheers

@robertblust

**Comment by norbertspiess** [comment]
> which UML Diagram you refer here?
> 
> @robertblust

I mean the [model.png](https://github.com/3AP-AG/candidate-assignment-java/blob/develop/model.png). You clearly stated that relationships are not added on purpose. But just the visual alignment irritated me, I guess. But having a closer look at the data / clarifying talk about it would have solved this immediately.

File: src/main/java/ch/aaap/assignment/model/PostalCommunity.java (hunk 1 above)
Comment (norbertspiess) [reply src/main/java/ch/aaap/assignment/model/PostalCommunity.java:9]: I identified the zipCode + zipCodeAddition uniqueness, but for simplicity reasons just went for an equals on all properties (as I didn't need a uniqueness of the postal communities, as they were created only once anyway).
I updated it nonethless.

**Comment by robertblust** [comment]
> I mean the [model.png](https://github.com/3AP-AG/candidate-assignment-java/blob/develop/model.png). You clearly stated that relationships are not added on purpose. But just the visual alignment irritated me, I guess. But having a closer look at the data / clarifying talk about it would have solved this immediately.

OK got it. The final model looks like this

model

If you like to play around, we have here an API which exposes the Data based on the model
https://platform-prod.3ap.ch/api/swagger/swagger-ui/index.html?configUrl=https%3A%2F%2Fplatform-prod.3ap.ch%2Fapi%2Fswagger%2Fconfig&urls.primaryName=zipcode-service

**Pull request closed without merging**